conda-store-server/conda_store_server/server/views/api.py

```diff
@@ -346,7 +354,234 @@ async def api_update_namespace(
         return {"status": "ok"}
 
 
-@router_api.delete("/namespace/{namespace}/", response_model=schema.APIAckResponse)
+@router_api_v2.put("/namespace/{namespace}/metadata", response_model=schema.APIAckResponse)  # fmt: skip
+async def api_update_namespace_metadata(
+    namespace: str,
+    request: Request,
+    metadata: Dict[str, Any] = None,
+    auth=Depends(dependencies.get_auth),
+    conda_store=Depends(dependencies.get_conda_store),
+):
+    with conda_store.get_db() as db:
+        auth.authorize_request(
+            request,
+            namespace,
+            {
+                Permissions.NAMESPACE_UPDATE,
+            },
+            require=True,
+        )
```

Comment:
The previous version of this PR had a bug, where the required permissions were not matching the purpose of the handler because I reused the handler body in a common function. In this version of the PR, each handler is separate.

Reply:
I also wonder if we need to have `NAMESPACE_READ` here and in other ones because we call `get_namespace` first.